cmd/wego/app/add/cmd.go

```diff
@@ -84,15 +84,33 @@ func runCmd(cmd *cobra.Command, args []string) error {
 	osysClient := osys.New()
 
 	token, err := osysClient.GetGitProviderToken()
-	if err != nil {
-		return err
+
+	if err == osys.ErrNoGitProviderTokenSet {
+		// No provider token set, we need to do the auth flow.
+		url := params.Url
+		if url == "" {
+			// Find the url using an unauthenticated git client. We just need to read the URL.
+			// params.Dir must be defined here because we already checked for it above.
+			url, err = git.New(nil).GetRemoteUrl(params.Dir, "origin")
+			if err != nil {
+				return fmt.Errorf("could not get remote url for directory %s: %w", params.Dir, err)
+			}
+		}
+		// DoAppRepoCLIAuth will take over the CLI and block until the flow is complete.
+		token, err = app.DoAppRepoCLIAuth(url, os.Stdout)
```

Comment:
The shadow variable is still here, can you change this to `authErr`.  I can tell you from experience that these side effects are a pain to debug.

Reply:
Please see my other comment on the PR regarding when shadowing is ok.